addons/web/controllers/main.py

```diff
@@ -665,8 +665,8 @@ def create(self, master_pwd, name, lang, password, **post):
     @http.route('/web/database/duplicate', type='http', auth="none", methods=['POST'], csrf=False)
     def duplicate(self, master_pwd, name, new_name):
         try:
-            if not re.match('^[a-zA-Z][a-zA-Z0-9_]+$', new_name):
-                raise Exception(_('Invalid database name. Only alphanumerical characters and underscore are allowed.'))
+            if not re.match('^[a-zA-Z0-9][a-zA-Z0-9_.-]+$', new_name):
+                raise Exception(_('Invalid database name. Only alphanumerical characters, underscore, hyphen and dot are allowed.'))
```

Comment:
Since both regex literal and message are used multiple times, maybe they should be extracted to constants, and injected into the rendering of `database_manager.html` (`Database._render_template`) so they don't have to also be duplicated there?